src/patternfly/components/Tabs/tabs.scss

```diff
@@ -1,106 +1,254 @@
 
   position: relative;
   display: flex;
-  flex-direction: column;
-  align-items: flex-start;
+  padding-right: var(--pf-c-tabs--Inset);
+  padding-left: var(--pf-c-tabs--Inset);
+  overflow: hidden;
 
-  // Scroll buttons hidden by default
-  .pf-c-tabs__scroll-button {
-    display: none;
-    visibility: hidden;
+  &::before {
+    border-color: var(--pf-c-tabs--before--BorderColor);
+    border-style: var(--pf-c-tabs--before--BorderStyle);
+    border-width: var(--pf-c-tabs--before--BorderTopWidth) var(--pf-c-tabs--before--BorderRightWidth) var(--pf-c-tabs--before--BorderBottomWidth) var(--pf-c-tabs--before--BorderLeftWidth);
   }
 
-  // Scroll buttons shown with modifiers
-  &.pf-m-start .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end .pf-c-tabs__scroll-button:nth-of-type(2) {
-    display: block;
-    visibility: visible;
+  // Filled style
+  &.pf-m-fill {
+    .pf-c-tabs__list {
+      flex-basis: 100%;
+    }
+
+    .pf-c-tabs__item {
+      flex-grow: 1;
+    }
+
+    .pf-c-tabs__button {
+      flex-basis: 100%;
+      justify-content: center;
+    }
   }
 
-  // Scroll button current state
-  &.pf-m-start-current:not(.pf-m-tabs-secondary) .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end-current:not(.pf-m-tabs-secondary) .pf-c-tabs__scroll-button:nth-of-type(2) {
-    &::after {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      margin-left: var(--pf-c-tabs__item--BorderWidth);
-      content: "";
-      border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--m-current--Color);
+  // Scroll buttons enabled
+  &.pf-m-scrollable {
+    .pf-c-tabs__scroll-button {
+      opacity: 1;
+    }
+
+    // Scroll buttons
+    .pf-c-tabs__scroll-button:nth-of-type(1) {
+      margin-right: 0;
+      transform: translateX(0);
+    }
+
+    .pf-c-tabs__scroll-button:nth-of-type(2) {
+      margin-left: 0;
+      transform: translateX(0);
+    }
+
+    &.pf-m-secondary,
+    &.pf-m-no-border {
+      --pf-c-tabs--before--BorderWidth: 0;
     }
   }
 
-  // Scroll button current state
-  &.pf-m-start-current.pf-m-tabs-secondary .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end-current.pf-m-tabs-secondary .pf-c-tabs__scroll-button:nth-of-type(2) {
-    color: var(--pf-c-tabs__scroll-button--m-secondary-right--m-start-current--Color);
+  // Remove bottom border for variants
+  &.pf-m-box,
+  &.pf-m-vertical {
+    .pf-c-tabs__button {
+      --pf-c-tabs__button--after--BorderBottomWidth: 0;
+    }
   }
 
-  // Primary with secondary tabs style
-  + &.pf-m-tabs-secondary {
-    margin-top: calc(-1 * var(--pf-c-tabs__item--BorderWidth));
-    border-top: var(--pf-c-tabs__item--BorderWidth) solid var(--pf-c-tabs__item--BorderColor);
+  // Box
+  &.pf-m-box {
+    --pf-c-tabs__button--BackgroundColor: var(--pf-c-tabs--m-box__button--BackgroundColor);
+    --pf-c-tabs__button--before--BorderBottomWidth: var(--pf-c-tabs__button--before--BorderWidth);
```

Comment:
if you end up removing those vars above, maybe here you could just set the border width

Reply:
You would update `--pf-c-tabs__button--before--BorderWidth` that would affect all `::before` borders. Here I'm just saying, enable the bottom border.